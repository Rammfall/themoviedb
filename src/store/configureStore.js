import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
