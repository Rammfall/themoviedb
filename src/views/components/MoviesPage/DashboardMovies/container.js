import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import MoviesList from 'Views/components/MoviesList'
import { getDashboardMoviesSelector } from 'Store/concepts/dashboard/selectors'

const DashboardMovies = ({ movies, total }) => (
  <MoviesList
    movies={movies}
    total={total}
  />
)

DashboardMovies.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({
    original_title: PropTypes.string,
    id: PropTypes.number,
    overview: PropTypes.string,
    poster_path: PropTypes.string
  })).isRequired,
  total: PropTypes.number
}

DashboardMovies.defaultProps = {
  total: null
}

const mapStateToProps = (state) => ({
  movies: getDashboardMoviesSelector(state).movies,
  total: getDashboardMoviesSelector(state).total
})

export { DashboardMovies as DashboardMoviesContainer }
export default connect(mapStateToProps)(DashboardMovies)
