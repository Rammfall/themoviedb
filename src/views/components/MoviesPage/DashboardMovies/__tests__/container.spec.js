import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import DashboardMovies, { DashboardMoviesContainer } from '../container'

jest.mock('Store/concepts/dashboard/selectors', () => ({
  getDashboardMoviesSelector: jest.fn(() => ({
    movies: [{
      id: 1,
      name: 'test',
      description: 'test'
    }],
    total: 1
  }))
}))

describe('DashboardMovies', () => {
  const store = configureStore()()

  it('matches snapshot', () => {
    const wrapper = shallow(<DashboardMovies store={store} />)
    const container = diveTo(wrapper, DashboardMoviesContainer)

    expect(container).toMatchSnapshot()
  })
})
