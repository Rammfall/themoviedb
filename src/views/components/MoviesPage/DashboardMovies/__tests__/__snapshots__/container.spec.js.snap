// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardMovies matches snapshot 1`] = `
<MoviesList
  actions={null}
  movies={
    Array [
      Object {
        "description": "test",
        "id": 1,
        "name": "test",
      },
    ]
  }
  total={1}
/>
`;
